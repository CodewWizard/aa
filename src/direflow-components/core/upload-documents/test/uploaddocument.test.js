import React from "react";
import renderer from "react-test-renderer";
import UploadDocument from "../uploaddocument";
import { render, screen, cleanup, fireEvent, waitFor } from "@testing-library/react";

afterEach(()=>{
    cleanup();    
})

const tableRowData = [
  {
    TransactionType: "Manual Upload",
    DocumentType: "Data Submission",
    TransactionDate: "June 15th, 2023 05:13:18 AM",
    EffectiveDate: "June 15th, 2023 05:13:18 AM",
    Uri: "https://devdiep2.azure-api.net/dms/CNGA/M/TX/TOGACNMTX230306/Quote_Prem%20Full%20Test_1.pdf",
  },
  {
    TransactionType: "Application",
    DocumentType: "Submission Data Document",
    TransactionDate: "June 15th, 2023 04:20:53 AM",
    EffectiveDate: "June 15th, 2023 04:20:53 AM",
    Uri: "https://devdiep2.azure-api.net/dms/CNGA/M/TX/TOGACNMTX230306/Quote_Prem%20Full%20Test_1.pdf",
  },
];

const UploadDocumentSchema = [
  {
    key: "UploadDocument",
    type: "Card",
    title: "Manual Upload",
    className: "container",
    titleclassName: "commonHead commonHeadPadding",
    childsClassName: "boxWrapper border-0 px-3 commonShadow",
    controls: [
      {
        key: "PolicyNumber",
        label: "Policy Number",
        type: "text",
        dataPath: "Policy.PolicyNumber",
        props: {
          className: "inputText",
          placeholder: "Policy Number",
          required: true,
          disabled: true,
        },
        positionClass: "col-md-6",
        isColummnDisplay: false,
      },
      {
        key: "DocumentType",
        label: "Document Type",
        type: "text",
        dataPath: "Policy.DocumentType",
        props: {
          className: "inputText",
          placeholder: "Document Type",
          required: true,
        },
        positionClass: "col-md-6",
        isColummnDisplay: false,
      },
      {
        key: "UploadFile",
        label: "Upload File",
        type: "file",
        props: {
          required: true,
          class: "inputText",
          placeholder: "Enter file",
          min: "0",
          max: "100",
          step: "10",
        },
        labelClass: "col-md-4",
        positionClass: "col-md-12 mt-3",
        isColummnDisplay: true,
        errorMessage: "Please enter valid file",
        dataPath: "Policy.UploadFile",
      },
      {
        key: "filerules",
        label: "",
        type: "staticContent",
        value: `<p>* Maximum file size - 100 MB</p>
          <p>** Allowed Formats - jpeg, jpg, png, gif, bmp, xls, xlsx, doc, docx, odt, txt, pdf, pptx, ppt, rtf, eml, csv</p>`,
        props: {
          class: "textBlack",
        },
        labelClass: "col-md-4",
        inputClass: "col-md-12 mt-3",
        positionClass: "col-md-12 mt-3",
        isColummnDisplay: true,
        isOverlayTrigger: true,
      },
    ],
  },
];

const UploadDocumentModalForm = {
  schema: UploadDocumentSchema,
  submitLabel: "Upload",
  dataModel: {},
  onSchemaChange: (model, key) => {},
  onFormSubmit: (dataModel, event) => console.log(dataModel),
};

const EmailDocumentSchema = [
  {
    key: "EmailDocument",
    type: "Card",
    title: "Email Document",
    className: "container",
    titleclassName: "commonHead commonHeadPadding",
    childsClassName: "boxWrapper border-0 px-3 commonShadow",
    controls: [
      {
        key: "EmailID",
        label: "Email ID (Enter comma separated email address)",
        type: "text",
        dataPath: "Email.EmailID",
        props: {
          className: "inputText",
          placeholder: "Enter Email Address",
          required: true,
        },
        positionClass: "col-md-12 mt-2",
        isColummnDisplay: false,
      },
      {
        key: "OptionalContent",
        label: "Add optional email content",
        type: "textarea",
        dataPath: "Email.EmailBody",
        props: {
          className: "inputText",
          placeholder: "Enter Optional Email Body(Max 1000 Characters)",
        },
        positionClass: "col-md-12 mt-4",
        isColummnDisplay: false,
      },
    ],
  },
];

const EmailDocumentModalForm = {
  submitLabel: "Email",
  schema: EmailDocumentSchema,
  dataModel: {},
  onSchemaChange: (model, key) => {},
  onFormSubmit: (dataModel, event, Uri) => {
    console.log([dataModel, event, Uri]);
  },
  onError: () => {},
};

const tableCols = [
  {
    dataField: "TransactionType",
    text: "Status",
    sort: false,
    headerStyle: () => {
      return { width: "20%", textAlign: "center" };
    },
    customFormat: (data) => {
      return data + "-OLD";
    },
  },
  {
    dataField: "DocumentType",
    text: "DOCUMENT TYPE",
    sort: false,
    headerStyle: () => {
      return { width: "20%", textAlign: "center" };
    },
  },
  {
    dataField: "TransactionDate",
    text: "Uploaded / Created Date",
    sort: false,
    headerStyle: () => {
      return { width: "25%", textAlign: "center" };
    },
  },
  {
    dataField: "Uri",
    text: "FileUrl",
    sort: false,
    headerStyle: () => {
      return { width: "20%", textAlign: "center" };
    },
    hidden: true,
  },
];

const CommonArgs = {
  title: "Documents",
  tableCols,
  tableKey: "TransactionType",
  tableRowData,
  UploadDocumentModalForm,
  EmailDocumentModalForm,
  isExpandRow: true,
  groupByColumn: "TransactionType",
};

it("should render given schema correctly", () => {
  const testRenderer = renderer.create(<UploadDocument {...CommonArgs} />);
  expect(testRenderer.toJSON()).toMatchSnapshot();
  testRenderer.unmount();
});

it("should contain 'Upload Document' button", () => {
  render(<UploadDocument {...CommonArgs} />);
  const uploadDocumentBtn = screen.getByText(/Upload Document/g);
  expect(uploadDocumentBtn).not.toBeNull();
});

it("should display modal popup when 'Upload Document' button is clicked", async () => {
  render(<UploadDocument {...CommonArgs} />);
  const uploadDocumentBtn = screen.getByText(/Upload Document/g);
  fireEvent.click(uploadDocumentBtn);
  await waitFor(() => {
    const closeBtn = screen.getByText(/Close/g);
    expect(closeBtn).not.toBeNull();
  });

});





