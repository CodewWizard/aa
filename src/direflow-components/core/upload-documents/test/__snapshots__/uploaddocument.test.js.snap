// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render given schema correctly 1`] = `
Array [
  <div
    className="first-table position-relative"
  >
    <h3
      className="m-0"
    >
      Documents
       Â 
    </h3>
    <div
      className="summaryEdit"
    >
      <a
        className="edit-2"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-upload fa-lg "
          data-icon="upload"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M288 109.3V352c0 17.7-14.3 32-32 32s-32-14.3-32-32V109.3l-73.4 73.4c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l128-128c12.5-12.5 32.8-12.5 45.3 0l128 128c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L288 109.3zM64 352H192c0 35.3 28.7 64 64 64s64-28.7 64-64H448c35.3 0 64 28.7 64 64v32c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V416c0-35.3 28.7-64 64-64zM432 456a24 24 0 1 0 0-48 24 24 0 1 0 0 48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Upload Document
      </a>
    </div>
    <div
      className="w-100"
    >
      <div
        className="singleBox customDataTable"
      >
        <div
          className="react-bootstrap-table"
        >
          <table
            className="table"
          >
            <thead>
              <tr>
                <th
                  style={
                    Object {
                      "textAlign": "center",
                      "width": "20%",
                    }
                  }
                  tabIndex={0}
                >
                  Status
                </th>
                <th
                  style={
                    Object {
                      "textAlign": "center",
                      "width": "20%",
                    }
                  }
                  tabIndex={0}
                >
                  DOCUMENT TYPE
                </th>
                <th
                  style={
                    Object {
                      "textAlign": "center",
                      "width": "25%",
                    }
                  }
                  tabIndex={0}
                >
                  Uploaded / Created Date
                </th>
                <th
                  className="expand-cell-header"
                  data-row-selection={true}
                  onClick={[Function]}
                >
                  (+)
                </th>
              </tr>
            </thead>
            <tbody>
              <tr
                className=""
                onClick={[Function]}
                style={Object {}}
              >
                <td>
                  Manual Upload
                </td>
                <td>
                  Data Submission
                </td>
                <td>
                  June 15th, 2023 05:13:18 AM
                </td>
                <td
                  className="expand-cell"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-angle-down fa-lg "
                    data-icon="angle-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M201.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 274.7 86.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </td>
              </tr>
              <tr
                className=""
                onClick={[Function]}
                style={Object {}}
              >
                <td>
                  Application
                </td>
                <td>
                  Submission Data Document
                </td>
                <td>
                  June 15th, 2023 04:20:53 AM
                </td>
                <td
                  className="expand-cell"
                  onClick={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-angle-down fa-lg "
                    data-icon="angle-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M201.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 274.7 86.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>,
  <a
    alt="filedownload"
    id="filedownload"
  />,
]
`;
